service/worker/scheduler/workflow_test.go

```diff
@@ -1402,3 +1405,80 @@ func (s *workflowSuite) TestLimitedActions() {
 	s.True(s.env.IsWorkflowCompleted())
 	// doesn't end properly since it sleeps forever after pausing
 }
+
+// To generate workflow history for a specific server version, run the server with that version
+// uncomment the first line to run the workflow with the given schedule, and run the test. Make sure
+// to set the timeout flag accordingly. When test is finished, run temporal client with the command at the end
+// to capture the JSON history.
+func (s *workflowSuite) TestGenerateWorkflowHistoryForReplay() {
+	//s.T().Skip("Remove this Skip to generate the history for a specific version.")
+	c, _ := client.Dial(client.Options{})
+	defer c.Close()
+
+	sid := "sched-test-replay"
+	wid := "sched-test-replay-wf"
+	tq := "sched-test-tq"
+
+	var runs int32
+	workflowFn := func(ctx workflow.Context) error {
+		workflow.SideEffect(ctx, func(ctx workflow.Context) any {
+			atomic.AddInt32(&runs, 1)
+			return runs
+		})
+		return nil
+	}
+	w := worker.New(c, tq, worker.Options{})
+	w.RegisterWorkflow(workflowFn)
+
+	_ = w.Start()
+	defer w.Stop()
+
+	ctx := context.Background()
+	numberOfActionsToTake := 50
+	interval := 1
+	scheduleHandle, err := c.ScheduleClient().Create(ctx, client.ScheduleOptions{
+		ID: sid,
+		Spec: client.ScheduleSpec{
+			Intervals: []client.ScheduleIntervalSpec{
+				{
+					Every: time.Duration(interval) * time.Second,
+				},
+			},
+		},
+		Action: &client.ScheduleWorkflowAction{
+			ID:        wid,
+			Workflow:  workflowFn,
+			TaskQueue: tq,
+		},
+		Overlap:          enumspb.SCHEDULE_OVERLAP_POLICY_ALLOW_ALL,
+		RemainingActions: numberOfActionsToTake,
+	})
+	s.NoError(err)
+	deleteScheule := func() {
+		// delete schedule
+		err = scheduleHandle.Delete(ctx)
+		s.NoError(err)
+	}
+
+	defer deleteScheule()
+
+	s.Eventually(func() bool { return atomic.LoadInt32(&runs) == int32(numberOfActionsToTake) }, time.Duration((numberOfActionsToTake+2)*interval)*time.Second, time.Duration(interval)*time.Second)
```

Comment:
ideally, do some stuff like pause in the middle, unpause later, do an update.

also using buffer_all with some workflows that sleep for a few seconds (to create a buffer) will provide greater coverage.

actually, now that I'm writing that, these are basically all cases that the integration test does (or should do.. some aren't implemented yet, and we should add another for the getnexttime cache that does 20 iterations). maybe we should have a script that runs the integration test and then grabs the histories for all the schedules it used?

Reply:
hmm, I think that's a better idea. I also think the replay test should not be a unit-test since it's not testing a unit of work. Maybe we can place it along with the integration tests?

Reply:
the replay test is definitely a unit test. it is testing a unit of code in isolation, specifically the workflow implementation. the thing that generates the histories to test with is not a test at all, though.